Domains ui

hectorsanchezmx opened the pull request:
This UI replaces the previous text field to update the instance domain, to allow users to manage the domains their instance is mapped to.

Three main things:
- Adds Table component: the idea was to make it as independent as possible to consider it for the design system. It may need some tweaks for that.
- Adds the UI and store logic for retrieving, adding and deleting domain names.

![domains-ui](https://user-images.githubusercontent.com/3080957/78930300-79b0b500-7a69-11ea-8e2f-8978aee23b25.gif)

shrunyan commented:
Looks great. Couple of notes.

- Please change "save" button to specific "**Add Domain**"
- Add sentence below "Add a new domain" title: "**Enter your domain and select the code branch you want to be served from the domain**"  _Might not be clear what the branch selection is doing_

shrunyan commented:
Couple more notes.

- Constrict the modals to a reasonable width. 
- Remove the original "Domain" title as this is now redundant with the table columns
- Match the table left alignment of the users table. _Notice the users flush left relative to the tiles and icons._
- "Add Domain" button should have a prefixed "plus" icon

shrunyan commented:
Also realizing we need to account for the new instance "launch" flow which directs the user to this view. https://accounts.zesty.io/instances/8-f48cf3a682-7fthvk/launch

hectorsanchezmx commented:
@shrunyan latest updates are in the branch now, they include:
- Moved New Domain modal logic to separate component
- Updated the Table styles using grid (calculates the amount of columns based on headers)
- Updated the  LaunchWizard component to use the New Domain Modal
- Added condition to show on Settings the LaunchWizard if there are no custom domains setup yet

shrunyan commented:
- Launch component displays twice when clicking the "Launch Instance" button
- Once a domain is added in the Launch component nothing is updated indicating the domain is set.
  - The Launch component domain input experience should have stayed the same but update the logic on which API it connects to. No need to open a modal when an inline input reduces the amount of clicks.
- *app crashes when navigating around instance settings. Bad function reference ```TypeError: "_this2.props.domains.filter is not a function"```
- I think the modal for domain input is unnecessary. It should just be inline on the card. Same way we layout adding new user access. ```domain | branch | add button```

hectorsanchezmx commented:
@shrunyan I undid some of the changes before your last comments and added the following:

- Updated the. Domain component to make it an inline form instead of a modal
- Updated the condition to show the LaunchWizard

shrunyan commented:
- adding and removing domains should be limited to the Owner and Admin roles.
- clear add domain fields after successfully adding the domain
- `/launch` should always render the wizard. Once a domain is added from this experience it should remain rendered and then the domain listing card should render below it.

hectorsanchezmx commented:
@shrunyan pushed the following updates:

- Added the condition to show only the input for admins
- Cleared the domain input after submitting the form to add a new domain
- Launch wizard sets default branch to Live
- Updated conditional rendering to show on /launch to always show the Launch Wizard and only show the table when there are some domains set already

shrunyan commented:
Can you change the phrasing on the domain card to: 
> You must be this instances owner or admin to manage domains

shrunyan commented on src/apps/properties/src/views/PropertyOverview/PropertyOverview.js:
```diff
@@ -36,17 +38,34 @@ class PropertyOverview extends Component {
+        const customDomain = domainParts
+          .slice(Math.max(domainParts.length - 2, 0))
+          .join('.')
+        return customDomain !== 'zesty.dev'
```
We also need to account for the prior `.zesty.site` domain.
